compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -2508,31 +2548,31 @@ abstract class GenJSCode extends plugins.PluginComponent
       val expr = genExpr(selector)
       val resultType = toIRType(tree.tpe)
 
-      val List(defaultBody0) = for {
-        CaseDef(Ident(nme.WILDCARD), EmptyTree, body) <- cases
-      } yield body
-
-      val (defaultBody, defaultLabelSym) = defaultBody0 match {
-        case LabelDef(_, Nil, rhs) if hasSynthCaseSymbol(defaultBody0) =>
-          (rhs, defaultBody0.symbol)
-        case _ =>
-          (defaultBody0, NoSymbol)
-      }
-
-      val genDefaultBody = genStatOrExpr(defaultBody, isStat)
+      val defaultLabelSym = cases.collectFirst {
+        case CaseDef(Ident(nme.WILDCARD), EmptyTree,
```

Comment:
That would still be a `WILDCARD`, and in the body there would be a `val n: Int = x`. You can see that we already were only handling WILDCARD here: https://github.com/scala-js/scala-js/pull/2543/files/595ddc57fa831f88ae3a42b8916832417f8b79a6#diff-c9a195a739d51194e79f0655bc93bd61R2615